pkg/deploy/deploy.go

```diff
@@ -359,6 +358,9 @@ func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstr
 	found.SetGroupVersionKind(obj.GroupVersionKind())
 	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
 	if err != nil {
+		if strings.Contains(err.Error(), "no matches for kind") {
+			return nil, nil // ignore mising CRD error
```

Comment:
should we just handle the notfound directly in the getResource()?
and we 
e.g 
```
func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
	found := &unstructured.Unstructured{}
	// Setting gvk is required to do Get request
	found.SetGroupVersionKind(obj.GroupVersionKind())
	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
	if err != nil {
		if errors.Is(err, &meta.NoKindMatchError{}) {
			return nil, nil // ignore mising CRD error
		}
		if client.IgnoreNotFound(err) != nil {
			return nil, nil // not found resource
		}
		return nil, err
	}
	return found, nil
}
```
for the caller:
```
	// Return if error getting resource in cluster
	found, err := getResource(ctx, cli, obj); 
	if err != nil {
		return err
	}
```

Reply:
This will work, but I do not like the idea to return nil object with nil error. It's prone to mistakes. But I do not have a strong opinion. Let's ask one more point of view? :) @bartoszmajsak

Reply:
@zdtsw @ykaliuta In my opinion, returning `nil, nil` can cause more harm than good. The response should either be the requested item or an error. This way, you only need to check for `nil` in the case of an error, allowing you to handle exceptions appropriately. Otherwise, we risk cluttering the codebase with `nil` checks or encountering nil pointer errors at runtime, as the caller may assume the returned instance is valid if the error is `nil`. Therefore, I believe it is preferable to return `apierrs.NewNotFound` as @ykaliuta suggests.

Reply:
follow up in https://github.com/opendatahub-io/opendatahub-operator/pull/1045 to address comments.